### PR TITLE
[LogisticRegression] Padding zeros to bypass the out-of-bound memory access of the raft stats (sum, mean, stddev) kernels

### PR DESCRIPTION
**Pull request by lijinf2**
This PR is a workaround to bypass gpu memory out-of-bound access of raft 24.02.

### CODE REVIEW

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/classification.py [hunk 1]
```diff
@@ -979,9 +979,33 @@ def _logistic_regression_fit(
             if self.getStandardization() is True and is_sparse is True:
                 concated = concated.toarray()
 
+            num_pad_zero_cols = 0
+            if self.getStandardization() is True and concated.shape[1] % 32 != 0:
+                num_pad_zero_cols = concated.shape[1] % 32
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/classification.py:984]: Shouldn't the number of padding columns be 32 - concated.shape[1] % 32 ?

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/classification.py (hunk 1 above)
Comment (lijinf2) [reply python/src/spark_rapids_ml/classification.py]: Good catch!

File: python/src/spark_rapids_ml/classification.py [hunk 2]
```diff
@@ -979,9 +979,35 @@ def _logistic_regression_fit(
             if self.getStandardization() is True and is_sparse is True:
                 concated = concated.toarray()
 
+            num_pad_zero_cols = 0
```
Comment (eordentlich) [inline python/src/spark_rapids_ml/classification.py:982]: Maybe add a comment about why this is done and that it is temporary work around.

**Review by eordentlich — COMMENTED** [review]
Looks good.  One more minor comment.

**Comment by lijinf2** [comment]
build

File: python/src/spark_rapids_ml/classification.py (hunk 2 above)
Comment (lijinf2) [reply python/src/spark_rapids_ml/classification.py:985]: Sounds good. Added a comment.

**Comment by lijinf2** [comment]
build

**Review by eordentlich — APPROVED** [review]
👍

**Pull request merged**